### PR TITLE
cluster list and remove updates to be composable

### PR DESCRIPTION
**Pull request by kensipe**
* list clusters ids `dcos cluster list --quiet` or `dcos cluster list -q`
* remove a list of ids
* composable `dcos cluster remove $(dcos cluster list --quiet)`
* better documentation

### CODE REVIEW

**Comment by kensipe** [comment]
The CI integration tests are NOT setup like the developer test instructions... there is no `DCOS_DIR` setup so these tests fail.   Analyzing the jenkins file and testing... it does not appear that the cluster directory structure (under the DCOS_URL) is ever tested in the CI env tests.    It appears that the `dcos.snakeoil.mesosphere.com` is associated with the DCOS_URL in the /etc/hosts file and that all tests are configured to use the `tests/data/dcos.toml`

**Comment by kensipe** [comment]
@klueska all good here with integration tests

**Comment by bamarni** [comment]
https://github.com/kensipe/dcos-cli/blob/2a6419c78fcd1314d9feae2bb1e7d96f598da53a/cli/dcoscli/data/help/cluster.txt#L26

Could we make it clear here that the remove command now accepts multiple clusters? eg. `Remove configured cluster(s) from the CLI.`

**Comment by kensipe** [comment]
rebased and updated the help text as requested.

File: cli/tests/integrations/test_cluster.py [hunk 1]
```diff
@@ -27,6 +31,67 @@ def test_list():
...
+
+    assert num_of_clusters() == 0
+
+
+def num_of_clusters():
+    _, stdout, _ = exec_command(
+        ['dcos', 'cluster', 'list', '--json'])
+    cluster_list = json.loads(stdout.decode('utf-8'))
+    return len(cluster_list)
+
```
Comment (klueska) [inline cli/tests/integrations/test_cluster.py]: Isn't this already done elsewhere? I'm sure there are other tests that rely on this. We should think about putting this in a helper library instead of duplicating it all over the place.

File: dcos/cluster.py [hunk 2]
```diff
@@ -215,9 +215,9 @@ def get_clusters():
 
 def get_cluster(name):
     """
-    :param name: name of cluster
+    :param name: name or cluster_id of cluster
     :type name: str
-    :returns: Cluster identified by name
+    :returns: Cluster identified by name or cluster_id
```
Comment (klueska) [inline dcos/cluster.py]: These changes to appear to logically below to this PR. I'm hesitant to squash this into the commit since it is unrelated.

**Comment by kensipe** [comment]
rebased... I will address nits later today

**Comment by kensipe** [comment]
rebased

File: cli/dcoscli/data/help/cluster.txt [hunk 3]
```diff
@@ -50,6 +50,8 @@ Options:
         Specify the authentication provider to use for login.
     --private-key=<key_path>
         Specify the path to a file that contains the private key.
+    -q, --quiet
+        Only display cluster IDs
```
Comment (elezar) [inline cli/dcoscli/data/help/cluster.txt]: Here we refer to `IDs`, but it seems as if we refer to them as `names` elsewhere.

**Comment by kensipe** [comment]
rebased and addressed @elezar feedback

**Comment by kensipe** [comment]
run linux integration tests

**Comment by kensipe** [comment]
@bamarni ready to roll!

**Comment by bamarni** [comment]
@kensipe : as we already have #1015 in the pipeline, I'd rather leave this one on hold. It's less user-friendly and we don't have any other use-case for composition including the cluster list right now.

This feature can be great when one is also able to filter using criterias (similarly to `docker ps` used with `docker rm` or `docker stop`). So we might revisit it in the future.

**Comment by kensipe** [comment]
@bamarni @klueska I think this is a mistake... I know there were 2 ways presented to potentially remove all clusters... however this way is the "best" way IMO and adds more... it adds the ability to make things composable.   I would want this even with #1015 added to the code base.

**Comment by bamarni** [comment]
@kensipe : could you please elaborate on why it is better? I'm not closing the door forever here, just saying that I don't see any use-case for this at the moment (as we are going to add the `--all` flag to `dcos cluster remove`).
